src/ordering/service.py

```diff
@@ -1,8 +1,53 @@
+import logging
 from uuid import UUID
 
+from injector import inject
+
+from application.settings import Settings
+from currency import BTC, ExchangeRateService
+
 from .commands import CreateBuyOrder
+from .db import Repository
+from .errors import BalanceLimitExceeded, OrderAlreadyExists
+from .events import BuyOrderCreated
 
+log = logging.getLogger(__name__)
 
+
+@inject
 class Service:
+    def __init__(
+            self,
+            settings: Settings,
+            repository: Repository,
+            exchange_rates: ExchangeRateService,
+    ) -> None:
+        self._bought_btc_limit = settings.ordered_btc_limit
```

Comment:
Why pass entire settings here? Why `Service` (or all other classes for that matter) need to know the entire `Settings`?

Alternatively, you could be passing what you need from settings directly to `Service` initializer, which would make it simpler (as it doesn't have to know about settings)

Reply:
fixed

Reply:
:+1: